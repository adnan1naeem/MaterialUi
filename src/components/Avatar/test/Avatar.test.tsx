import * as React from 'react';
import { Avatar } from '../';
import renderer from 'react-test-renderer';
import FolderIcon from 'material-ui-icons/Folder';

test('Avatar component with size props', () => {
	const avatar = renderer.create(
      <Avatar size={100}></Avatar>
	).toJSON();
	expect(avatar.props.style).toEqual([{ 'width':100, 'height':100, backgroundColor:'lightgrey', borderRadius:1000, display:'flex', justifyContent:'center', alignItems:'center', fontSize:.5*100 },{}]);
	expect(avatar).toMatchSnapshot();
});

test('Avatar component with style', () => {
	const avatar = renderer.create(
      <Avatar style={{ backgroundColor:'green' }}></Avatar>
  ).toJSON();
	expect(avatar?avatar.props.style:{}).toEqual([{ 'width':70, 'height':70, backgroundColor:'lightgrey', borderRadius:1000, display:'flex', justifyContent:'center', alignItems:'center', fontSize:.5*70 },{ backgroundColor:'green' }]);
	expect(avatar).toMatchSnapshot();
});

test('Avatar component with src props', () => {
	const avatar = renderer.create(
      <Avatar src="https://material-ui-next.com/static/images/remy.jpg"></Avatar>
  ).toJSON();
	expect(avatar.props.source).toEqual('https://material-ui-next.com/static/images/remy.jpg');
	expect(avatar).toMatchSnapshot();
});


test('Avatar component with src and size props', () => {
	const avatar = renderer.create(
      <Avatar src="https://material-ui-next.com/static/images/remy.jpg" size={200}></Avatar>
  ).toJSON();
	expect(avatar.props.style).toEqual([{ 'width':200, 'height':200, backgroundColor:'lightgrey', borderRadius:1000, display:'flex', justifyContent:'center', alignItems:'center', fontSize:.5*200 },{}]);
	expect(avatar.props.source).toEqual('https://material-ui-next.com/static/images/remy.jpg');

	expect(avatar).toMatchSnapshot();
});



test('Avatar component with children props', () => {
	const avatar = renderer.create(
      <Avatar size={200}>hello</Avatar>
	).toJSON();
	expect(avatar.props.children).toEqual(undefined);
	expect(avatar).toMatchSnapshot();
});
