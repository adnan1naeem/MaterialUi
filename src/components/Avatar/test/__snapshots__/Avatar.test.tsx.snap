// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component with children props 1`] = `
<div
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgrey",
        "borderRadius": 1000,
        "display": "flex",
        "fontSize": 100,
        "height": 200,
        "justifyContent": "center",
        "width": 200,
      },
      Object {},
    ]
  }
>
  hello
</div>
`;

exports[`Avatar component with size props 1`] = `
<img
  source={undefined}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgrey",
        "borderRadius": 1000,
        "display": "flex",
        "fontSize": 50,
        "height": 100,
        "justifyContent": "center",
        "width": 100,
      },
      Object {},
    ]
  }
/>
`;

exports[`Avatar component with src and size props 1`] = `
<img
  source="https://material-ui-next.com/static/images/remy.jpg"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgrey",
        "borderRadius": 1000,
        "display": "flex",
        "fontSize": 100,
        "height": 200,
        "justifyContent": "center",
        "width": 200,
      },
      Object {},
    ]
  }
/>
`;

exports[`Avatar component with src props 1`] = `
<img
  source="https://material-ui-next.com/static/images/remy.jpg"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgrey",
        "borderRadius": 1000,
        "display": "flex",
        "fontSize": 35,
        "height": 70,
        "justifyContent": "center",
        "width": 70,
      },
      Object {},
    ]
  }
/>
`;

exports[`Avatar component with style 1`] = `
<img
  source={undefined}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgrey",
        "borderRadius": 1000,
        "display": "flex",
        "fontSize": 35,
        "height": 70,
        "justifyContent": "center",
        "width": 70,
      },
      Object {
        "backgroundColor": "green",
      },
    ]
  }
/>
`;
